be/src/storage/rowset/segment_iterator.cpp

```diff
@@ -469,6 +522,158 @@ Status SegmentIterator::_init() {
+    CHECK(hit_indexes.size() <= 1) << "Only support query no more than one index now";
+
+    if (hit_indexes.empty()) {
+        return Status::OK();
+    }
+
+    auto tablet_index_meta = std::make_shared<TabletIndex>(hit_indexes[0]);
+    // TODO: to support below when tenann support this interface
+    // tenann::IndexMeta::Deserialize(tablet_index_meta.properties_to_json());
+
+    // std::string index_path = IndexDescriptor::vector_index_file_path(_opts.rowset_path, _opts.rowsetid.to_string(),
+    //                                                                  segment_id(), tablet_index_meta->index_id());
+
+    return _init_reader_from_file("index_path", tablet_index_meta);
+}
+
+Status SegmentIterator::_get_row_ranges_by_vector_index() {
+    RETURN_IF(!_use_vector_index, Status::OK());
+    RETURN_IF(_scan_range.empty(), Status::OK());
+
+    SCOPED_RAW_TIMER(&_opts.stats->get_row_ranges_by_vector_index_timer);
+
+    Status st;
+    std::map<rowid_t, float> id2distance_map;
+    std::vector<int64_t> result_ids;
+    std::vector<float> result_distances;
+    std::vector<int64_t> filtered_result_ids;
+    DelIdFilter del_id_filter(_scan_range);
+
+    {
+        SCOPED_RAW_TIMER(&_opts.stats->vector_search_timer);
+        if (_vector_range > 0) {
+            st = _ann_reader->range_search(_query_view, _k, &result_ids, &result_distances, &del_id_filter,
+                                           static_cast<float>(_vector_range), _result_order);
+            if (!st.ok()) {
+                // fall over to normal search and host range search to column expression predicate
+                LOG(WARNING) << "Vector index search failed with range_serach: " << st.to_string();
+                result_ids.resize(_k);
+                result_distances.resize(_k);
+                st = _ann_reader->search(_query_view, _k, (result_ids.data()),
+                                         reinterpret_cast<uint8_t*>(result_distances.data()), &del_id_filter);
+            }
+        } else {
+            result_ids.resize(_k);
+            result_distances.resize(_k);
+            st = _ann_reader->search(_query_view, _k, (result_ids.data()),
+                                     reinterpret_cast<uint8_t*>(result_distances.data()), &del_id_filter);
+        }
+    }
+
+    if (!st.ok()) {
+        LOG(WARNING) << "Vector index search failed: " << st.to_string();
+        return Status::InternalError(st.to_string());
+    }
+    SCOPED_RAW_TIMER(&_opts.stats->process_vector_distance_and_id_timer);
+
+    for (size_t i = 0; i < result_ids.size() && result_ids[i] != -1; i++) {
+        id2distance_map[result_ids[i]] = result_distances[i];
+    }
+
+    SparseRange r;
+    RETURN_IF_ERROR(_get_row_ranges_by_row_ids(&result_ids, &r));
+
+    size_t prev_size = _scan_range.span_size();
+    _scan_range = _scan_range.intersection(r);
+    _opts.stats->rows_stats_filtered += (prev_size - _scan_range.span_size());
```

Comment:
`rows_stats_filtered` is used for `ZoneMapIndexFilterRows`, should we use a new profile field like `VectorIndexFilterRows` here?